numpyro/distributions/continuous.py

```diff
@@ -67,6 +66,117 @@
+
+    @lazy_property
+    def right_scale(self):
+        return self.scale / self.asymmetry
+
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
+        z = value - self.loc
+        z = -jnp.abs(z) / jnp.where(z < 0, self.left_scale, self.right_scale)
+        return z - jnp.log(self.left_scale + self.right_scale)
+
+    def sample(self, key, sample_shape=()):
+        assert is_prng_key(key)
+        shape = (2,) + sample_shape + self.batch_shape + self.event_shape
+        u, v = random.exponential(key, shape=shape)
+        return self.loc - self.left_scale * u + self.right_scale * v
+
+    @property
+    def mean(self):
+        total_scale = self.left_scale + self.right_scale
+        return self.loc + (self.right_scale**2 - self.left_scale**2) / total_scale
+
+    @property
+    def variance(self):
+        left = self.left_scale
+        right = self.right_scale
+        total = left + right
+        p = left / total
+        q = right / total
+        return p * left**2 + q * right**2 + p * q * total**2
+
+    def cdf(self, value):
+        return jnp.where(
+            value <= self.loc,
+            self.asymmetry**2
+            / (1 + self.asymmetry**2)
+            * jnp.exp(self.right_scale * (value - self.loc)),
+            1
+            - 1
+            / (1 + self.asymmetry**2)
+            * jnp.exp(-self.left_scale * (value - self.loc)),
+        )
+
+    def icdf(self, value):
+        temp = self.asymmetry**2 / (1 + self.asymmetry**2)
+        return jnp.where(
+            value <= temp,
+            self.loc + self.left_scale * jnp.log(value / temp),
+            self.loc
+            - self.right_scale * (jnp.log1p(self.asymmetry**2) + jnp.log1p(-value)),
+        )
+
+
+class AsymmetricLaplaceQuantile(AsymmetricLaplace):
+    """An alternative parameterization of AsymmetricLaplace commonly applied in
+    Bayesian quantile regression.
+
+    Instead of the `asymmetry` parameter employed by AsymmetricLaplace, to
+    define the balance between left- versus right-hand sides of the
+    distribution, this class utilizes a `quantile` parameter, which describes
+    the proportion of probability density that falls to the left-hand side of
+    the distribution.
+
+    The `scale` parameter is also interpreted slightly differently than in
+    AsymmetricLaplce. When `loc=0` and `scale=1`, AsymmetricLaplace(0,1,1)
+    is equivalent to Laplace(0,1), while AsymmetricLaplaceQuantile(0,1,0.5) is
+    equivalent to Laplace(0,2).
+    """
+
+    arg_constraints = {
+        "loc": constraints.real,
+        "scale": constraints.positive,
+        "quantile": constraints.interval(0.0, 1.0),
+    }
+    reparametrized_params = ["loc", "scale"]
+    support = constraints.real
+
+    def __init__(self, loc=0.0, scale=1.0, quantile=0.5, validate_args=None):
+        asymmetry = jnp.sqrt(1 / ((1 / quantile) - 1))
+        scale = scale * asymmetry / quantile
```

Comment:
Do you think this will cause confusion - we init with some scale but the distribution stores another scale? I think it is best to keep scale as-is.

Reply:
Makes sense. This was something I originally experimented with when trying to have ALDQ inhert from ALD, but I think that broke tests when attempting to call super init methods. Will revert back to keeping a `self._ald = AsymmetricLaplace(...)` and then calling `self._ald` methods for `log_prob`, `cdf`, etc. I've been struggling to install the development environment using pip, it keeps breaking on scipy installation, but I think I've got it fixed now, so should be able to run tests locally until they pass before pushing more changes.